src/SdkCommon/AppAuthentication/Azure.Services.AppAuthentication.Unit.Tests/AzureServiceTokenProviderTests.cs

```diff
@@ -141,6 +141,66 @@ public void InvalidAzureAdInstanceTest()
             Assert.Contains(Constants.MustUseHttpsError, exception.ToString());
         }
 
+        /// <summary>
+        /// If resource id is null or empty, an exception should be thrown. 
+        /// </summary>
+        [Fact]
+        public async Task ResourceNullOrEmptyWhenGettingTokenTest()
+        {
+            // Mock ProcessManager is being asked to act like Azure CLI is not installed. 
+            MockProcessManager mockProcessManager = new MockProcessManager(MockProcessManager.MockProcessManagerRequestType.ProcessNotFound);
+            AzureCliAccessTokenProvider azureCliAccessTokenProvider = new AzureCliAccessTokenProvider(mockProcessManager);
+            AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider(azureCliAccessTokenProvider);
+
+            var exception = await Assert.ThrowsAsync<ArgumentNullException>(() => Task.Run(() => azureServiceTokenProvider.GetAccessTokenAsync(null, Constants.TenantId)));
+
+            Assert.Contains(Constants.CannotBeNullError, exception.ToString());
+
+            exception = await Assert.ThrowsAsync<ArgumentNullException>(() => Task.Run(() => azureServiceTokenProvider.GetAccessTokenAsync(string.Empty, Constants.TenantId)));
+
+            Assert.Contains(Constants.CannotBeNullError, exception.ToString());
+
+        }
+
+        /// <summary>
+        /// If azureAdInstance is null or empty, an exception should be thrown. 
+        /// </summary>
+        [Fact]
+        public void NullOrEmptyAzureAdInstanceTest()
+        {
+            var exception = Assert.Throws<ArgumentNullException>(() => new AzureServiceTokenProvider(azureAdInstance: null));
+
+            Assert.Contains(Constants.CannotBeNullError, exception.ToString());
+
+            exception = Assert.Throws<ArgumentNullException>(() => new AzureServiceTokenProvider(azureAdInstance: string.Empty));
+
+            Assert.Contains(Constants.CannotBeNullError, exception.ToString());
+        }
+
+        /// <summary>
+        /// If connectionstring is not specified in AzureServiceTokenProvider, ensure connection string is assigned from 
+        /// AzureServicesAuthConnectionString environment variable.
+        /// </summary>
+        [Fact]
+        public void UnspecifiedConnectionStringTest()
+        {
+            string connString = Environment.GetEnvironmentVariable("AzureServicesAuthConnectionString");
+            
+            if (connString == null)
+            {
+                // Set environment variable AzureServicesAuthConnectionString
+                Environment.SetEnvironmentVariable("AzureServicesAuthConnectionString", Constants.AzureCliConnectionString);
+                // Assert variable exists
```

Comment:
There is already a constant for the env name. Also, you do not need to get the variable and check for existence. You can just set it. Also, do add code to remote env variable in Dispose method. Check the MSITokenProviderTests.cs file.